public/app/features/alerting/unified/hooks/useFilteredRules.ts

```diff
@@ -1,35 +1,103 @@
-import { useMemo } from 'react';
+import produce from 'immer';
+import { compact } from 'lodash';
+import { useCallback, useEffect, useMemo } from 'react';
 
 import { getDataSourceSrv } from '@grafana/runtime';
-import { useQueryParams } from 'app/core/hooks/useQueryParams';
-import { CombinedRuleGroup, CombinedRuleNamespace, FilterState } from 'app/types/unified-alerting';
-import { PromRuleType, RulerGrafanaRuleDTO } from 'app/types/unified-alerting-dto';
+import { Matcher } from 'app/plugins/datasource/alertmanager/types';
+import { CombinedRuleGroup, CombinedRuleNamespace } from 'app/types/unified-alerting';
+import { isPromAlertingRuleState, PromRuleType, RulerGrafanaRuleDTO } from 'app/types/unified-alerting-dto';
 
-import { labelsMatchMatchers, parseMatchers } from '../utils/alertmanager';
+import { getSearchFilterFromQuery, SearchFilterState, applySearchFilterToQuery } from '../search/searchParser';
+import { labelsMatchMatchers, matcherToMatcherField, parseMatcher, parseMatchers } from '../utils/alertmanager';
 import { isCloudRulesSource } from '../utils/datasource';
-import { getFiltersFromUrlParams } from '../utils/misc';
-import { isAlertingRule, isGrafanaRulerRule } from '../utils/rules';
+import { isAlertingRule, isGrafanaRulerRule, isPromRuleType } from '../utils/rules';
 
-export const useFilteredRules = (namespaces: CombinedRuleNamespace[]) => {
-  const [queryParams] = useQueryParams();
-  const filters = getFiltersFromUrlParams(queryParams);
+import { useURLSearchParams } from './useURLSearchParams';
 
-  return useMemo(() => {
-    const filteredNamespaces = namespaces
-      // Filter by data source
-      // TODO: filter by multiple data sources for grafana-managed alerts
+export function useRulesFilter() {
+  const [queryParams, updateQueryParams] = useURLSearchParams();
+  const searchQuery = queryParams.get('search') ?? '';
+
+  const filterState = getSearchFilterFromQuery(searchQuery);
+  const hasActiveFilters = Object.values(filterState).some((fs) => (Array.isArray(fs) ? fs.length > 0 : !!fs));
```

Comment:
I think we can shorten this to `Boolean(fs?.length)`? Can it be something other than `string` or `Array`?

Reply:
what about using the `isEmpty` from Lodash?

Reply:
Good ideas!